core/src/main/java/com/linecorp/armeria/server/logging/ContentPreviewingServiceBuilder.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server.logging;
+
+import static com.google.common.base.Preconditions.checkState;
+
+import java.nio.charset.Charset;
+import java.util.function.Function;
+
+import com.linecorp.armeria.common.logging.ContentPreviewerFactory;
+import com.linecorp.armeria.common.logging.ContentPreviewingDecoratorBuilder;
+import com.linecorp.armeria.server.HttpService;
+
+/**
+ * Builds a new {@link ContentPreviewingService} or its decorator function.
+ */
+public final class ContentPreviewingServiceBuilder extends ContentPreviewingDecoratorBuilder {
```

Comment:
Don't we need a package-private default constructor?

Reply:
Oops we need that. Thanks!